PCTL add seems to ignore that path variable specified in a nested profile.

<!--
Thank you for opening this bug report! You are helping make pctl a better project :)
In order to help us process this issue faster, please provide as much detail as possible.

When providing command examples, error output or code samples, please use code blocks.
For verbose output, for example logs, please either create a gist or use html details tags.
-->

**What were you trying to accomplish?**
<!-- Information about the command you ran and what you expected to happen. -->

I am trying to install a nested from profile inside a profile.

**What happened?**
<!-- A description of actual behavior (with error messages). -->

It seems that is generating the profile from the source location, but when it imports the child profile is using the "." path location instead of the one specified in the profile location.

► generating profile installation from source: repository git@github.com:weaveworksREDACTIED.git, path: ./REDACTIED and branch demo-profile
✗ failed to make artifacts: failed to read profile.yaml in repo "git@github.com:weaveworks/REDACTIED.git" branch "main" path ".": open /var/folders/l5/z__sm53j3bs2p5znngsf96kc0000gn/T/cloned_profiledec0c6924581307/profile.yaml: no such file or directory
make: *** [test] Error 1

    - name: $NAME
      profile:
        source:
          url: git@github.com:weaveworks/{REDACTIED}.git
          tag: main
          path: ./REDACTIED

**How to reproduce it?**
<!--
Include ALL the steps to reproduce the bug.

If using a config file, include it here, removing any sensitive information!
-->

Please reach out individual to me over slack for actual configuration file.

**Logs**
<!--
Include the output of the command line when running pctl.
Make sure you redact any sensitive information before posting.
If the output is long, please consider a Gist, or an html details tag.
-->

► generating profile installation from source: repository git@github.com:weaveworks/REDACTIED.git, path: ./REDACTIED and branch demo-profile
✗ failed to make artifacts: failed to read profile.yaml in repo "git@github.com:weaveworks/REDACTIED.git" branch "main" path ".": open /var/folders/l5/z_

**Anything else we need to know?**
<!--
What OS are you using?
Are you using a downloaded binary or did you compile pctl?
-->
Mac OS

**Versions**
<!--
What pctl version are you using?
-->
pctl version 0.11.0

--- a/pkg/install/installer.go
+++ b/pkg/install/installer.go
@@ -78,8 +78,7 @@ func (i *Installer) collectArtifacts(installation profilesv1.ProfileInstallation
 	for _, a := range profileDef.Spec.Artifacts {
 		//If its a nested profile lets make a recursive call to scans its artifacts
 		if a.Profile != nil {
-			err := validateProfileArtifact(a.Profile)
-			if err != nil {
+			if err := validateProfileArtifact(a.Profile); err != nil {
 				return nil, err
 			}
 			nestedInstallation := installation.DeepCopyObject().(*profilesv1.ProfileInstallation)

--- a/pkg/install/installer.go
+++ b/pkg/install/installer.go
@@ -78,6 +78,10 @@ func (i *Installer) collectArtifacts(installation profilesv1.ProfileInstallation
 	for _, a := range profileDef.Spec.Artifacts {
 		//If its a nested profile lets make a recursive call to scans its artifacts
 		if a.Profile != nil {
+			err := validateProfileArtifact(a.Profile)
+			if err != nil {
+				return nil, err
+			}
 			nestedInstallation := installation.DeepCopyObject().(*profilesv1.ProfileInstallation)
 			nestedInstallation.Spec.Source.URL = a.Profile.Source.URL
 			nestedInstallation.Spec.Source.Branch = a.Profile.Source.Branch
@@ -108,6 +112,17 @@ func (i *Installer) collectArtifacts(installation profilesv1.ProfileInstallation
 		}
 	}
 	return artifacts, nil
+}
+
+func validateProfileArtifact(p *profilesv1.Profile) error {
+	if p.Source.Tag != "" && p.Source.Branch != "" {
+		return fmt.Errorf("cannot configure both %q and %q in profile artifact", "profile.Source.Tag", "Profile.Source.Branch")
+	}
+
+	if p.Source.Tag == "" && p.Source.Branch == "" {
+		return fmt.Errorf("one of %q or %q must be configured", "profile.Source.Tag", "Profile.Source.Branch")
+	}
+	return nil
 }
 
 func (i *Installer) cloneRepoAndGetProfileDefinition(repoURL, branch, path string) (profilesv1.ProfileDefinition, error) {

--- a/pkg/install/installer_test.go
+++ b/pkg/install/installer_test.go
@@ -215,6 +215,25 @@ var _ = Describe("Installer", func() {
 			Expect(err).To(MatchError(ContainSubstring("failed to parse profile.yaml")))
 		})
 	})
+
+	When("the nested profile is invalid", func() {
+		When("branch and tag are both configured", func() {
+			It("returns an error", func() {
+				profileDefinition1.Spec.Artifacts[1].Profile.Source.Tag = "foo"
+				err := installer.Install(installation)
+				Expect(err).To(MatchError(ContainSubstring("cannot configure both \"profile.Source.Tag\" and \"Profile.Source.Branch\" in profile artifact")))
+			})
+		})
+
+		When("neither branch and tag are both configured", func() {
+			It("returns an error", func() {
+				profileDefinition1.Spec.Artifacts[1].Profile.Source.Branch = ""
+				profileDefinition1.Spec.Artifacts[1].Profile.Source.Tag = ""
+				err := installer.Install(installation)
+				Expect(err).To(MatchError(ContainSubstring("one of \"profile.Source.Tag\" or \"Profile.Source.Branch\" must be configured")))
+			})
+		})
+	})
 })
 
 func writeKubernetesResource(obj runtime.Object, dir, filename string) error {

